### ISSUE TITLE
TypeError: Cannot read property 'statusCode' of undefined

### ISSUE THREAD

**Issue by askmike** [aarongarvey/binance#31]
Here is the full error:

    **[PATH REDACTED]**\gekko\node_modules\binance\lib\rest.js:78
                        if (response.statusCode === 400 && payload.code === -1021 &&
                                     ^

    TypeError: Cannot read property 'statusCode' of undefined
        at Request.request [as _callback] (**[PATH REDACTED]**\gekko\node_modules\binance\lib\rest.js:78:34)
        at self.callback (**[PATH REDACTED]**\gekko\node_modules\request\request.js:186:22)
        at Request.emit (events.js:160:13)
        at ClientRequest.<anonymous> (**[PATH REDACTED]**\gekko\node_modules\request\request.js:816:16)
        at Object.onceWrapper (events.js:255:19)
        at ClientRequest.emit (events.js:160:13)
        at TLSSocket.emitTimeout (_http_client.js:708:34)
        at Object.onceWrapper (events.js:255:19)
        at TLSSocket.emit (events.js:160:13)
        at TLSSocket.Socket._onTimeout (net.js:412:8)

cross post from https://github.com/askmike/gekko/issues/1781.

**Comment by aarongarvey** [comment]
Were you using version 1.3.1 when you got this?  The code in your stack trace, on line 78, is no longer on line 78 but line 80 so I'm going to assume not?  The change in your pull request assumes there's a case where both `err` and `response` are undefined.  Has that been your experience?

I was assuming one or the other would always be defined.  I attempted to fix the error you mentioned here https://github.com/aarongarvey/binance/commit/06573afd0e401c089ab1223716d655b1a93a1511, as it was reported yesterday https://github.com/aarongarvey/binance/issues/26.

**Comment by askmike** [comment]
Ah, I didn't see that this was fixed yesterday. This might have been under
1.3.0, which gekko might install (see here:
https://github.com/askmike/gekko/blob/develop/package.json ). I will update
my dependencies and report back if your fix turned out not sufficiant.

Regards,
Mike

On 24 Jan 2018 21:07, "Aaron Garvey" <notifications@github.com> wrote:

Were you using version 1.3.1 when you got this? The code in your stack
trace, on line 78, is no longer on line 78 but line 80 so I'm going to
assume not? The change in your pull request assumes there's a case where
both err and response are undefined. Has that been your experience?

I was assuming one or the other would always be defined. I attempted to fix
the error you mentioned here 06573af
<https://github.com/aarongarvey/binance/commit/06573afd0e401c089ab1223716d655b1a93a1511>,
as it was reported yesterday #26
<https://github.com/aarongarvey/binance/issues/26>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/aarongarvey/binance/issues/31#issuecomment-360257629>,
or mute the thread
